src/core/utils/useOnlineInternalHandleLoanReservation.ts

```diff
@@ -0,0 +1,112 @@
+import { useQueryClient } from "react-query";
+import {
+  getGetV1UserLoansQueryKey,
+  getGetV1UserReservationsQueryKey,
+  usePostV1UserLoansIdentifier,
+  usePostV1UserReservationsIdentifier
+} from "../../core/publizon/publizon";
+import { usePatronData } from "../../core/utils/helpers/usePatronData";
+import useReaderPlayer from "../../core/utils/useReaderPlayer";
+import { useUrls } from "../../core/utils/url";
+import { useModalButtonHandler } from "../../core/utils/modal";
+import { onlineInternalModalId } from "../../apps/material/helper";
+import { getAllFaustIds } from "../../core/utils/helpers/general";
+import { Manifestation } from "../../core/utils/types/entities";
+import { OnlineInternalRequestStatus } from "../../core/utils/types/request";
+import { CreateLoanResult } from "../publizon/model";
+
+type useOnlineInternalHandleLoanReservationType = {
+  manifestations: Manifestation[];
+  openModal: boolean;
+  setReservationStatus?: (status: OnlineInternalRequestStatus) => void;
+  setLoanResponse?: (response: CreateLoanResult | null) => void;
+};
+
+const useOnlineInternalHandleLoanReservation = ({
+  manifestations,
+  openModal,
+  setReservationStatus,
+  setLoanResponse
+}: useOnlineInternalHandleLoanReservationType) => {
+  const queryClient = useQueryClient();
+  const u = useUrls();
+  const authUrl = u("authUrl");
+  const { openGuarded } = useModalButtonHandler();
+
+  const { mutate: mutateLoan } = usePostV1UserLoansIdentifier();
+  const { mutate: mutateReservation } = usePostV1UserReservationsIdentifier();
+  const { data: userData } = usePatronData();
+
+  const { canBeLoaned, canBeReserved, identifier } =
+    useReaderPlayer(manifestations);
+
+  const handleModalLoanReservation = () => {
+    if (openModal) {
+      openGuarded({
+        authUrl,
+        modalId: onlineInternalModalId(getAllFaustIds(manifestations))
+      });
+      return;
+    }
+
+    if (canBeLoaned && identifier) {
+      mutateLoan(
+        { identifier },
+        {
+          onSuccess: (res) => {
+            // Ensure that the button is updated after a successful loan
+            queryClient.invalidateQueries(getGetV1UserLoansQueryKey());
+            if (setReservationStatus) {
+              setReservationStatus("loaned");
+            }
+            if (setLoanResponse) {
+              setLoanResponse(res);
+            }
+          },
+          onError: () => {
+            if (setReservationStatus) {
+              setReservationStatus("error");
+            }
+          }
+        }
```

Comment:
I do not understand the relationship between the event handling (`onSuccess()` and `onError()`) and the state modifiers (`setReservationStatus()` and `setLoanResponse()`)

- Why do we set a reservation status when performing a loan? 
- Why do we set the response object for loans but a string for reservations?
- Why do se set a reservation status as error when a loan fails but do not set a loan status?